File: offgame-core/src/main/java/org/retal/offgame/dto/PlanetOverview.java
package org.retal.offgame.dto;

import lombok.*;

import java.time.Instant;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlanetOverview {

    private Instant serverTime;
    private String name;
    private String imageName;
    private Long diameter;
    private Long usedFields;
    private Long totalFields;
    private TechnologyOrderInfo activeTechnologyOrder;
    private BuildingOrderInfo activeBuildingOrder;
    private Long spaceYardTotalTimeLeft;
    private Long minTemperature;
    private Long maxTemperature;
    private Long galaxy;
    private Long system;
    private Long position;
}

File: offgame-ui/src/app/model/PlanetOverview.ts
import {TechnologyOrder} from "./TechnologyOrder";
import {BuildingOrder} from "./BuildingOrder";

export class PlanetOverview {

    serverTime!: string
    serverTimeDate: Date
    name!: string
    imageName!: string
    diameter!: number
    usedFields!: number
    totalFields!: number
    usedFieldsPercent: number
    activeTechnologyOrder?: TechnologyOrder
    activeBuildingOrder?: BuildingOrder
    spaceYardTotalTimeLeft?: number
    minTemperature!: number
    maxTemperature!: number
    galaxy!: number
    system!: number
    position!: number

    constructor(data: any) {
        Object.assign(this, data);

        this.usedFieldsPercent = this.usedFields / this.totalFields
        this.serverTimeDate = new Date(this.serverTime)
        if (data.activeTechnologyOrder) {
             this.activeTechnologyOrder = new TechnologyOrder(data.activeTechnologyOrder)
        }
        if (data.activeBuildingOrder) {
            this.activeBuildingOrder = new BuildingOrder(data.activeBuildingOrder)
        }
    }

    updateServerTime() {
        this.serverTimeDate = new Date(this.serverTimeDate.getTime() + 1000)
    }

    getCoordinates() {
        return `[${this.galaxy}:${this.system}:${this.position}]`
    }
}